### PR TITLE
Health check: optimistic status

### PR DESCRIPTION
**Pull request by terencechain**
Add health checks to cover optimistic cases. In this case, we return errors with message on corrective action

### CODE REVIEW

**Review by potuz — DISMISSED** [review]

File: beacon-chain/blockchain/service.go [hunk 1]
```diff
@@ -156,6 +156,15 @@ func (s *Service) Stop() error {
 // Status always returns nil unless there is an error condition that causes
 // this service to be unhealthy.
 func (s *Service) Status() error {
+	optimistic, err := s.IsOptimistic(s.ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if service is optimistic: %v", err)
+	}
+	if optimistic {
+		return fmt.Errorf("service is optimistic, and only limited service functionality is provided " +
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/service.go:164]:
```suggestion
		return errors.New("service is optimistic, and only limited service functionality is provided " +
```

File: beacon-chain/rpc/service.go [hunk 2]
```diff
@@ -382,6 +382,14 @@ func (s *Service) Stop() error {
 
 // Status returns nil or credentialError
 func (s *Service) Status() error {
+	optimistic, err := s.cfg.OptimisticModeFetcher.IsOptimistic(s.ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if service is optimistic: %v", err)
+	}
+	if optimistic {
+		return fmt.Errorf("service is optimistic, validators can't perform duties " +
```
Comment (prestonvanloon) [inline beacon-chain/rpc/service.go:390]:
```suggestion
		return errors.New("service is optimistic, validators can't perform duties " +
```

File: beacon-chain/rpc/service.go [hunk 3]
```diff
@@ -382,6 +382,14 @@ func (s *Service) Stop() error {
 
 // Status returns nil or credentialError
 func (s *Service) Status() error {
+	optimistic, err := s.cfg.OptimisticModeFetcher.IsOptimistic(s.ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if service is optimistic: %v", err)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/service.go:387]:
```suggestion
		return errors.Wrap(err, "failed to check if service is optimistic")
```

File: beacon-chain/blockchain/service.go [hunk 4]
```diff
@@ -156,6 +156,15 @@ func (s *Service) Stop() error {
 // Status always returns nil unless there is an error condition that causes
 // this service to be unhealthy.
 func (s *Service) Status() error {
+	optimistic, err := s.IsOptimistic(s.ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if service is optimistic: %v", err)
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/service.go:161]:
```suggestion
		return errors.Wrap(err, "failed to check if service is optimistic")
```

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

**Comment by rauljordan** [comment]
@prestonvanloon PTAL again

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**